### DISCUSSION TITLE
Change Port Number with prebuilt docker?

### DISCUSSION THREAD

**Discussion by dts350z** [JeffreyCA/spleeter-web#84] [Q&A] [answered]
Subject says it all. I already have stuff running on port 8000.

**Comment by JeffreyCA** [comment, marked as answer]
Good suggestion, added in https://github.com/JeffreyCA/spleeter-web/pull/86!

**Comment by keyosuke** [comment]
With the latest version that runs in 80, it runs locally but I can't share it with my friends, because my ISP blocks 80 and when I try to port redirect another external port to an internal 80 I get Bad Request (400), which makes me think I'm having problems with ALLOWED_HOSTS.  I've tried changing the port the server runs on back to 80, or to anything else, and haven't succeded, and I've tried adding my local IP address and external DNS names to ALLOWED_HOSTS in settings_docker.py, settings.py, and APP_HOST variable of .env file at different times but that didn't seem to help.

If I wanted to use the latest version and allow external traffic to see it on a different port than 80, is there a way?  I had that worked in the past but Demucs v3 without typing python commands is too alluring.  Also, is there a paypal where I can donate a few dollars for every time you respond?  I feel like this is the best software with great support, but in my business life I work with companies that accept millions of dollars for worse software that isn't supported as well.  At least let me buy you a drink.

**Reply by JeffreyCA** [reply]
What do you see in the Docker logs of the nginx and api containers when you hit the bad request? If you run the `docker-compose` command without `-d` it should stream the logs to stdout/err.

So if I understand correctly you want to access the app locally at port 80, but want others to access it through another port?

**Reply by JeffreyCA** [reply]
> Also, is there a paypal where I can donate a few dollars for every time you respond? I feel like this is the best software with great support, but in my business life I work with companies that accept millions of dollars for worse software that isn't supported as well. At least let me buy you a drink.

Thanks that means a lot🙂 I don't have one set up but I am waiting to get approved for GitHub Sponsors

**Reply by keyosuke** [reply]
I didn't see a -d in any of the commands I had typed to pull or start the docker containers, but while poking around I did another "docker-compose pull"  and the "docker-compose -f docker-compose.yml -f docker-compose.dev.yml up" and I started getting that restart loop again on the celerys

"
celery-fast_1  |   File "/webapp/api/tasks.py", line 15, in <module>
celery-fast_1  |     from .separators.demucs_separator import DemucsSeparator
celery-fast_1  |   File "/webapp/api/separators/demucs_separator.py", line 7, in <module>
celery-fast_1  |     from demucs.pretrained import load_pretrained, PRETRAINED_MODELS
celery-fast_1  | ImportError: cannot import name 'load_pretrained' from 'demucs.pretrained' (/usr/local/lib/python3.7/site-packages/demucs/pretrained.py)
celery-slow_1 exited with code 1
celery-fast_1 exited with code 1

"

I wonder what I'm doing to screw this up.  I think docker-compose pull fixed it the last time, but that doesn't seem to be doing it, this time, unless I'm missing some step.

**Reply by keyosuke** [reply]
I blew it all away and started from scratch, and dumped my old media folder in after doing the clone, pull, up, and now it's working, the issues with reaching it on my lan IP and DNS are all gone, and port 8000 is whjat it's running on, too.  I have no idea how, but I'm going to enjoy it.  

Looks like Demucs v3 is there, too, so it's not that I somehow cloned an old one, I don't think.

**Reply by JeffreyCA** [reply]
Glad to hear!

What's generally worked for me when "updating" Spleeter Web without losing data is:

1. `docker-compose -f ... -f ... down`
2. `git pull`
3. `docker-compose -f ... -f ... pull`
4. `docker system prune`
5. `docker-compose -f ... -f ... up`
